swap/src/network/spot_price.rs

```diff
@@ -59,11 +60,61 @@ pub enum Error {
         #[serde(with = "::bitcoin::util::amount::serde::as_sat")]
         buy: bitcoin::Amount,
     },
+    BlockchainNetworkMismatch {
+        cli: BlockchainNetwork,
+        asb: BlockchainNetwork,
+    },
     /// To be used for errors that cannot be explained on the CLI side (e.g.
     /// rate update problems on the seller side)
     Other,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
+pub struct BlockchainNetwork {
+    pub bitcoin: BitcoinNetwork,
+    pub monero: MoneroNetwork,
+}
```

Comment:
Not quite sure if modelling these out completely isn't just unnecessary complexity? Would be good to give this some proper thought eventually.

Reply:
Initially I had aggregated Bitcoin and Monero them into `Testnet` vs `Mainnet`, but I felt it somehwhat limits us in what networks are supported. I felt the network protocol should not make assumptions on how the application behaves -  on the application level we only support Bitcoin `testnet3` and Monero `stagenet` for the moment - but the network layer does not necessarily have to know that.
Happy to keep put more thought into this eventually...

Reply:
The main motivation for doing it this way is, that I wanted to keep the network protocol unaware of application implementation specifics. The fact that the current implementation only supports Monero `stagenet` and Bitcoin `testnet3` should not be a concern of the protocol...?

Initially I had aggregated both Bitcoin and Network networks in either `Testnet` or `Mainnet`, but I found that somewhat incorrect, because Monero is not Testnet but Stagenet :] 

I think the current solution is the most correct, but probably not the simplest